frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -53,40 +60,112 @@ export function MultiAutocomplete({
+
+    const values = parseValues(searchValue);
+    const validValues = values.filter(isValid);
+
     setSearchValue("");
+
+    if (validValues.length > 0) {
+      const newValues = unique([...lastSelectedValues, ...validValues]);
+      setSelectedValues(newValues);
+      setLastSelectedValues(newValues);
+    } else {
+      setSelectedValues(lastSelectedValues);
+    }
+
     onBlur?.(event);
   };
 
-  const handleSearchChange = (newSearchValue: string) => {
-    setSearchValue(newSearchValue);
+  const handlePaste = (event: ClipboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    const input = event.target as HTMLInputElement;
+    const value = input.value;
+    const before = value.slice(0, input.selectionStart ?? value.length);
+    const after = value.slice(input.selectionEnd ?? value.length);
 
-    const isValid = shouldCreate?.(newSearchValue, selectedValues);
-    if (isValid) {
-      setSelectedValues([...lastSelectedValues, newSearchValue]);
+    const pasted = event.clipboardData.getData("text");
+    const text = `${before}${pasted}${after}`;
+
+    const values = parseValues(text);
+    const validValues = values.filter(isValid);
+
+    if (values.length > 0) {
+      const newValues = unique([...lastSelectedValues, ...validValues]);
+      setSelectedValues(newValues);
+      setLastSelectedValues(newValues);
+      setSearchValue("");
     } else {
-      setSelectedValues(lastSelectedValues);
+      setSearchValue(text);
     }
   };
 
-  const handlePaste = (event: ClipboardEvent<HTMLInputElement>) => {
-    const text = event.clipboardData.getData("Text");
-    const values = text.split(/[\n,]/g);
-    if (values.length > 1) {
-      const validValues = [...new Set(values)]
-        .map(value => value.trim())
-        .filter(value => shouldCreate?.(value, selectedValues));
+  const handleSearchChange = (newSearchValue: string) => {
+    const first = newSearchValue.at(0);
+    const last = newSearchValue.at(-1);
+
+    setSearchValue(newSearchValue);
+
+    if (newSearchValue !== "") {
+      const values = parseValues(newSearchValue);
+      if (values.length === 1) {
+        const value = values[0] ?? newSearchValue;
+        if (value && shouldCreate?.(value, lastSelectedValues)) {
+          setSelectedValues(unique([...lastSelectedValues, value]));
+        }
+      }
+    }
+    if (newSearchValue === "") {
+      setSelectedValues(unique([...lastSelectedValues]));
+    }
+
+    const quotes = Array.from(newSearchValue).filter(ch => ch === '"').length;
+
+    if (
+      (last === "," && quotes % 2 === 0) ||
+      last === "\t" ||
+      last === "\n" ||
+      (first === '"' && last === '"')
+    ) {
+      const values = parseValues(newSearchValue);
+      const validValues = values.filter(
+        value => value !== "" && shouldCreate?.(value, lastSelectedValues),
+      );
+
+      if (values.length > 0) {
+        setSearchValue("");
+      }
```

Comment:
I think this is redundant - `setSearchValue("")` will happen if `validValues.length > 0` in the `if` block below